From 194a1b499c36db057319712220af8ec7400030d7 Mon Sep 17 00:00:00 2001
From: Lee, Chun-Yi <jlee@novell.com>
Date: Wed, 29 Dec 2010 00:17:23 +0800
Subject: [PATCH] Add Lenovo Ideapad S12 to acer blacklist

The Lenovo Ideapad S12's _WDG have GUID that matches with AMW0, it causes
acer-wmi driver auto-probe when system boot but there already have
ideapad-laptop driver to support Ideapad serial.
So, add Lenovo Ideapad S12 to acer blacklist to disable it.

Signed-off-by: Lee, Chun-Yi <jlee@novell.com>
Tested-by: Ville-Pekka Vainio <vpivaini@cs.helsinki.fi>
Acked-by: Carlos Corbacho <carlos@strangeworlds.co.uk>
Cc: Thomas Renninger <trenn@suse.de>
Cc: Dmitry Torokhov <dtor@mail.ru>
Cc: Corentin Chary <corentincj@iksaif.net>
Cc: Matthew Garrett <mjg59@srcf.ucam.org>
---
 drivers/platform/x86/acer-wmi.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

Index: linux-2.6.37/drivers/platform/x86/acer-wmi.c
===================================================================
--- linux-2.6.37.orig/drivers/platform/x86/acer-wmi.c
+++ linux-2.6.37/drivers/platform/x86/acer-wmi.c
@@ -228,6 +228,28 @@ static struct dmi_system_id __devinitdat
 			DMI_MATCH(DMI_PRODUCT_NAME, "AOA150"),
 		},
 	},
+	/*
+	 * The Lenovo Ideapad S12 have GUID that matches with AMW0 - disable it.
+	 * There already have ideapad-laptop driver to support Ideapad serial.
+	 */
+	{
+		.ident = "Lenovo Ideapad S12",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "20021,2959"),
+		},
+	},
+	/*
+	 * Lenovo 3000 v200 have GUID that matches with AMW0 - disable it as
+	 * it disables wireless
+	 */
+	{
+		.ident = "Lenovo 3000 V200",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "0764A11"),
+		},
+	},
 	{}
 };
 
